flocker/node/functional/test_docker.py

```diff
@@ -403,6 +403,68 @@ def test_pull_image_if_necessary(self):
         d.addCallback(lambda _: self.assertTrue(docker.inspect_image(image)))
         return d
 
+    def test_pull_timeout(self):
+        """
+        Pulling an image times-out if it takes longer than a provided timeout.
+        """
+        # Use an image that isn't likely to be in use by anything, since
+        # it's old, and isn't used by other tests:
+        image = u"ubuntu:12.04"
+        # Make sure image is gone:
+        docker = Client()
+        try:
+            docker.remove_image(image, force=True)
+        except APIError as e:
+            if e.response.status_code != 404:
+                raise
+
+        name = random_name(self)
+        client = DockerClient(
+            namespace=self.namespacing_prefix, long_timeout=1)
+        self.addCleanup(client.remove, name)
+        d = client.add(name, image)
+        # requests has a TimeoutError, but timeout raises a ConnectionError.
+        # Both are subclasses of IOError, so use that for now
+        # https://github.com/kennethreitz/requests/issues/2620
+        self.assertFailure(d, IOError)
+        return d
+
+    def test_pull_timeout_pull(self):
+        """
+        Image pull timeout does not affect subsequent pulls.
+        """
+        # Use an image that isn't likely to be in use by anything, since
```

Comment:
Instead, use an image specifically for this test (similar to https://github.com/ClusterHQ/flocker/pull/1281) - this has only a bit of overhead but is lower risk, and can be a tiny image and doesn't need this comment.